index.js

```diff
@@ -45,46 +46,27 @@ module.exports = class ServeDrive extends ReadyResource {
     return this.server.address()
   }
 
-  add (drive, opts = {}) {
-    if (opts.alias && opts.default) throw new Error('Can not use both alias and default')
-    if (drive.key === null) throw new Error('Drive is not ready')
-    if (!opts.default && !opts.alias && !drive.key) throw new Error('Localdrive needs an alias or to be the default')
+  getLink (path, id, version) {
+    const { port } = this.address()
 
-    if (opts.default) this.drives.set(null, drive)
-    if (opts.alias) this.drives.set(opts.alias, drive)
+    let link = `http://localhost:${port}/${path}`
+    if (id || version) link += '?'
+    if (id) link += `drive=${id}`
 
-    if (drive.key) this.drives.set(z32.encode(drive.key), drive)
+    if (id && version) link += '&'
+    if (version) link += `checkout=${version}`
+    return link
   }
 
-  delete (drive, opts = {}) {
-    if (opts.alias && opts.default) throw new Error('Can not use both alias and default')
-    if (!drive.opened && drive.key === null) throw new Error('Drive is not ready')
-    if (!opts.default && !opts.alias && !drive.key) throw new Error('Localdrive needs an alias or to be the default')
-
-    if (opts.default) this.drives.delete(null)
-    if (opts.alias) this.drives.delete(opts.alias)
-
-    if (drive.key) this.drives.delete(z32.encode(drive.key))
-  }
-
-  async _onrequest (req, res) {
-    if (req.method !== 'GET' && req.method !== 'HEAD') {
-      res.writeHead(400).end()
-      return
-    }
-
-    const { pathname, searchParams } = new URL(req.url, 'http://localhost')
-
-    const id = searchParams.get('drive') // String or null
-    const drive = this.drives.get(id)
-
+  async _driveToRequest (drive, req, res, pathname, { id, version }) {
```

Comment:
why the object for id and version?

Reply:
Unneeded indeed, made them normal params (as is it was bugged even, should have assigned to empty object by default)